### ISSUE TITLE
word2vec model missing tokens

### ISSUE THREAD

**Issue by dterg** [RaRe-Technologies/gensim#1549]
#### Description
Dictionary contains 12 million tokens, but word2vec model does not include all the tokens (even if they are present in the corpus more than the min_count). 

I am 'tagging' each token in the corpus with 2 codes after each token i.e.

[["**word1**", "C2034", "C3.203", "**word2**", "C00932", "C23.134"], ["**word3**", "C934", "C99.23.99" ....]]

Each unique token will have exactly the same codes following it in the corpus. In the dictionary, all the tokens including the codes are present (prune set to none to fit the 12 million tokens), but not in the model. Interestingly, all the **words** are in the model, but not the codes.

#### Steps/Code/Corpus to Reproduce
Rather than to reproduce (since the corpus is >15GB), this is the code and parameters I'm using
```
from gensim.models import word2vec

class MyCorpus(object):
      def __init__(self, dictionary):
              self.dictionary = dictionary

      def __iter__(self):
              with open('corpus_with_codes.txt','r', encoding='utf8') as readfile:
                       for line in readfile:
                             yield line.split()

print('Building dictionary')
dictionary = corpora.Dictionary((line.split() for line in open('corpus_with_codes.txt', encoding='UTF8')), prune_at=None)
corpus = MyCorpus(dictionary)

print('Training model')
model = Word2Vec(corpus, min_count=5, size=200, window=3, sg=1, workers=12)
print('Saving model')
model.wv.save_word2vec_format('model_with_codes.txt', binary=False)
```

#### Expected Results
For this given example: [["**word1**", "C2034", "C3.203", "**word2**", "C00932", "C23.134"], ["**word1**", "C934", "C99.23.99" ....]]

```
model.wv['word1'] --> 200 dim vector
model.wv['C2034'] --> 200 dim vector
model.wv['C3.203'] --> 200 dim vector
model.wv['word2'] --> 200 dim vector
...
```

#### Actual Results
```
model.wv['word1'] --> 200 dim vector
model.wv['C2034'] --> not in the model
model.wv['C3.203'] --> not in the model
model.wv['word2'] --> 200 dim vector
...
```

#### Package versions (Python 3.5.4 - conda env)
bleach==1.5.0
boto==2.48.0
bz2file==0.98
certifi==2017.7.27.1
chardet==3.0.4
**gensim==2.3.0**
html5lib==0.9999999
idna==2.6
Keras==2.0.7
Markdown==2.6.9
numpy==1.13.1
protobuf==3.4.0
PyYAML==3.12
requests==2.18.4
scipy==0.19.1
six==1.10.0
smart-open==1.5.3
tensorflow==1.3.0
tensorflow-gpu==1.3.0
tensorflow-tensorboard==0.1.4
Theano==0.9.0
urllib3==1.22

**Comment by gojomo** [comment]
If a key isn't in the model post-training, then almost certainly it wasn't in the text examples provided – at least not enough times to reach the `min_count` threshold. 

Your example of your annotated text differs from your description – 'word1' is shown paired with two different sets of codes. Should the second instance of 'word1' be considered some third word, to be more representative of the real data?

What does `MyCorpus` do? How can it provide individual texts (with adjacent words-and-code as I'm assuming is desired) if its input is a `corpus.Dictionary` that only contains the discovered tokens (mapped to int IDs)? (Note that it is *not* necessary to feed Word2Vec texts where tokens have been converted to int-IDs, or the text is converted into a bag-of-words vector. It expects lists-of-strings as its texts/sentences.)

Have you checked the first few items of the corpus iteration to see if they look as expected? (eg `list(itertools.islice(corpus,5)`)

**Comment by dterg** [comment]
I went through the text to make sure that the key I was looking for indeed exceeds the threshold of ```min_count```, which they did. This is expected since if each word is always coupled with the same codes, and the word is in the word2vec model, then the corresponding codes should also be in the model because of the 1:1 occurrence. In the text, I confirmed this 1:1 relationship for a number of tokens.

Correct, I've edited the second instance to 'word3'. Sorry about that. Also, I forgot to add the ```MyCorpus``` class. I will check the corpus iterator again but the results (excluding the fact the code tokens are missing) are reasonable. 

I'm suspecting it has to do with the window size, but I'm testing that at the moment.
